tests/test_phases.py

```
@@ -0,0 +1,463 @@
+
+
+def test_phase_recorder_exit_uses_current_thread_for_strict_thread_validation() -> None:
+    recorder = PhaseRecorder()
+    token, _boundary = recorder.enter(session_id="session-1", rank=0, name="train")
+    error: Exception | None = None
+
+    def _close_on_other_thread() -> None:
+        nonlocal error
+        try:
+            recorder.exit(token)
+        except Exception as exc:  # pragma: no branch - asserted below
```

Ruff (0.15.10)

[warning] Do not catch blind exception: `Exception`

(BLE001)